airbyte-cdk/python/airbyte_cdk/sources/declarative/auth/token_provider.py

```diff
@@ -0,0 +1,86 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+
+import datetime
+from abc import abstractmethod
+from dataclasses import InitVar, dataclass
+from typing import Any, List, Mapping, Optional, Union
+
+import dpath.util
+import pendulum
+import requests
+from airbyte_cdk.models import Level
+from airbyte_cdk.sources.declarative.decoders.decoder import Decoder
+from airbyte_cdk.sources.declarative.decoders.json_decoder import JsonDecoder
+from airbyte_cdk.sources.declarative.exceptions import ReadException
+from airbyte_cdk.sources.declarative.interpolation.interpolated_string import InterpolatedString
+from airbyte_cdk.sources.declarative.requesters.requester import Requester
+from airbyte_cdk.sources.declarative.types import Config
+from airbyte_cdk.sources.http_logger import format_http_message
+from airbyte_cdk.sources.message import MessageRepository, NoopMessageRepository
+from isodate import Duration
+from pendulum import DateTime
+
+
+class TokenProvider:
+    @abstractmethod
+    def get_token(self) -> str:
+        pass
+
+
+@dataclass
+class SessionTokenProvider(TokenProvider):
```

Comment:
Q: should we eventually update the OAuth authentication to also use a token provider? It looks like a similar pattern as the session token authenticator to me